src/app/cake/cake.service.ts

```diff
@@ -0,0 +1,29 @@
+import { Connection } from 'mongoose';
+
+import { DBService } from '../../components/database-service.component';
+import { CakeModel } from '../cake/cake.model';
+import { ICake, ICakeQuery } from '../cake/cake.interface';
+
+export class CakeService {
+    private dbService: DBService<ICake, ICakeQuery>;
+
+    constructor(conn: Connection) {
+        this.dbService = new DBService(conn, 'Cake', CakeModel);
+    }
+
+    get(id: string) {
```

Comment:
I recommend the return type `Promise<IBread>`

Reply:
On several places I'm getting this `'Promise<ICake>': [Symbol.toStringTag], finallyts(2739)` and I couldn't find a solution so far

Reply:
That's weird, never happen before. Did you try setting an array into the generic param? `Promise<ICake[]>`